tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,437 @@
+    }
+  }
+
+  // Test surface_normal_one_form
+  Strahlkorper<Frame::Inertial>::OneForm test_normal_one_form(n_pts);
+  {
+    const auto& r = strahlkorper.radius();
+    const auto temp = r + amp * sin_phi * sin_theta;
+    const auto one_over_r = 1.0 / r;
+    test_normal_one_form.get(0) = cos_phi * sin_theta * temp * one_over_r;
+    test_normal_one_form.get(1) =
+        (sin_phi * sin_theta * temp - amp) * one_over_r;
+    test_normal_one_form.get(2) = cos_theta * temp * one_over_r;
+  }
+  const auto& normal_one_form = strahlkorper.surface_normal_one_form();
+  for (size_t j = 0; j < 3; ++j) {
+    for (size_t s = 0; s < n_pts; ++s) {
+      CHECK(test_normal_one_form.get(j)[s] ==
+            approx(normal_one_form.get(j)[s]));
+    }
+  }
+
+  // Test surface_normal_magnitude.
+  Strahlkorper<Frame::Inertial>::InverseMetric invg(n_pts);
+  DataVector test_normal_mag(n_pts);
+
+  {
+    const auto& r = strahlkorper.radius();
+    invg.get(0, 0) = 1.0;
+    invg.get(1, 0) = 0.1;
+    invg.get(2, 0) = 0.2;
+    invg.get(1, 1) = 2.0;
+    invg.get(1, 2) = 0.3;
+    invg.get(2, 2) = 3.0;
+
+    // Nasty expression I computed in mathematica.
+    const auto normsquared =
+        (-0.30000000000000004 * cos_theta * (r + amp * sin_phi * sin_theta) *
+             (1. * amp * square(cos_phi) +
+              1. * amp * square(cos_theta) * square(sin_phi) -
+              1. * r * sin_phi * sin_theta +
+              cos_phi * (amp * (-0.6666666666666667 +
+                                0.6666666666666667 * square(cos_theta)) *
+                             sin_phi -
+                         0.6666666666666667 * r * sin_theta) +
+              cos_theta * (-10. * r - 10. * amp * sin_phi * sin_theta)) +
+         0.1 * cos_phi * (amp * (-1. + 1. * square(cos_theta)) * sin_phi -
+                          1. * r * sin_theta) *
+             (1. * amp * square(cos_phi) +
+              1. * amp * square(cos_theta) * square(sin_phi) -
+              1. * r * sin_phi * sin_theta +
+              cos_phi * (amp * (-10. + 10. * square(cos_theta)) * sin_phi -
+                         10. * r * sin_theta) +
+              cos_theta * (-2. * r - 2. * amp * sin_phi * sin_theta)) +
+         2. * (amp * square(cos_phi) +
+               sin_phi *
+                   (amp * square(cos_theta) * sin_phi - 1. * r * sin_theta)) *
+             (amp * square(cos_phi) +
+              amp * square(cos_theta) * square(sin_phi) -
+              1. * r * sin_phi * sin_theta +
+              cos_phi * (amp * (-0.05 + 0.05 * square(cos_theta)) * sin_phi -
+                         0.05 * r * sin_theta) +
+              cos_theta * (-0.15 * r - 0.15 * amp * sin_phi * sin_theta))) /
+        square(r);
+    test_normal_mag = sqrt(normsquared);
+  }
+  const auto& normal_mag = strahlkorper.surface_normal_magnitude(invg);
+  for (size_t s = 0; s < n_pts; ++s) {
+    CHECK(test_normal_mag[s] == approx(normal_mag[s]));
```

Comment:
We have a (relatively new) CHECK_ITERABLE_APPROX macro that does this loop for you.